### PR TITLE
Build and publish API and UI images

### PR DESCRIPTION
**Pull request by haikoschol**
- **refactor: Rename Worker Dockerfile and build step**
- **build: Add Dockerfile for API**
- **build: Build and push API image**
- **build: Add Dockerfile for UI**
- **build: Build and push UI image**

### CODE REVIEW

File: UI.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+FROM base AS deps
+WORKDIR /app/dos
+
+COPY package.json ./package.json
+COPY apps/clearance_ui/package.json ./apps/clearance_ui/package.json
+COPY packages/common-helpers/package.json ./packages/common-helpers/package.json
+COPY packages/s3-helpers/package.json ./packages/s3-helpers/package.json
+COPY package-lock.json ./package-lock.json
+
```
Comment (lamppu) [inline UI.Dockerfile:15]: Here's the package.json files that the UI will need:
`COPY package.json ./package.json
COPY apps/clearance_ui/package.json ./apps/clearance_ui/package.json
COPY packages/spdx-validation/package.json ./packages/spdx-validation/package.json
COPY packages/validation-helpers/package.json ./packages/validation-helpers/package.json
COPY package-lock.json ./package-lock.json`

File: API.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+FROM node:20.10.0
+
+WORKDIR /app/dos
+
+COPY package.json ./package.json
+COPY apps/api/package.json ./apps/api/package.json
+COPY packages/common-helpers/package.json ./packages/common-helpers/package.json
+COPY packages/s3-helpers/package.json ./packages/s3-helpers/package.json
+COPY package-lock.json ./package-lock.json
+
```
Comment (lamppu) [inline API.Dockerfile:15]: Here's the package.json files that the API will need:
COPY package.json ./package.json
COPY apps/api/package.json ./apps/api/package.json
COPY packages/database/package.json ./packages/database/package.json
COPY packages/s3-helpers/package.json ./packages/s3-helpers/package.json
COPY packages/validation-helpers/package.json ./packages/validation-helpers/package.json
COPY package-lock.json ./package-lock.json

**Comment by mmurto** [comment]
@haikoschol, [this](https://github.com/doubleopen-project/dos/compare/main...ui-docker-fixes) should get you a little further, now it fails at the env validation, that probably should be removed.

**Comment by haikoschol** [comment]
> @haikoschol, [this](https://github.com/doubleopen-project/dos/compare/main...ui-docker-fixes) should get you a little further, now it fails at the env validation, that probably should be removed.

Now it gets stuck again. 😔

```
[+] Building 1861.5s (13/19)                                                                                                                                                                                                                                      docker:orbstack
 => [internal] load build definition from UI.Dockerfile                                                                                                                                                                                                                      0.0s
 => => transferring dockerfile: 1.13kB                                                                                                                                                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/node:20.10.0                                                                                                                                                                                                              1.9s
 => [auth] library/node:pull token for registry-1.docker.io                                                                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                            0.0s
 => => transferring context: 220B                                                                                                                                                                                                                                            0.0s
 => [base 1/1] FROM docker.io/library/node:20.10.0@sha256:8d0f16fe841577f9317ab49011c6d819e1fa81f8d4af7ece7ae0ac815e07ac84                                                                                                                                                   0.0s
 => [internal] load build context                                                                                                                                                                                                                                            0.8s
 => => transferring context: 1.63MB                                                                                                                                                                                                                                          0.8s
 => CACHED [builder 1/4] WORKDIR /app                                                                                                                                                                                                                                        0.0s
 => [runner 2/6] RUN addgroup --system --gid 1001 nodejs                                                                                                                                                                                                                     0.8s
 => [builder 2/4] RUN npm i -g turbo                                                                                                                                                                                                                                         7.4s
 => [runner 3/6] RUN adduser --system --uid 1001 nextjs                                                                                                                                                                                                                      0.3s
 => [builder 3/4] COPY . .                                                                                                                                                                                                                                                   4.9s
 => [builder 4/4] RUN turbo prune clearance_ui --docker                                                                                                                                                                                                                      0.8s
 => [installer 2/5] COPY --from=builder /app/out/json/ .                                                                                                                                                                                                                     0.0s
 => [installer 3/5] RUN npm install                                                                                                                                                                                                                                       1845.6s
 => => # npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead
 => => # npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
```

**Comment by mmurto** [comment]
> > @haikoschol, [this](https://github.com/doubleopen-project/dos/compare/main...ui-docker-fixes) should get you a little further, now it fails at the env validation, that probably should be removed.
> 
> Now it gets stuck again. 😔
> 
> ```
> [+] Building 1861.5s (13/19)                                                                                                                                                                                                                                      docker:orbstack
>  => [internal] load build definition from UI.Dockerfile                                                                                                                                                                                                                      0.0s
>  => => transferring dockerfile: 1.13kB                                                                                                                                                                                                                                       0.0s
>  => [internal] load metadata for docker.io/library/node:20.10.0                                                                                                                                                                                                              1.9s
>  => [auth] library/node:pull token for registry-1.docker.io                                                                                                                                                                                                                  0.0s
>  => [internal] load .dockerignore                                                                                                                                                                                                                                            0.0s
>  => => transferring context: 220B                                                                                                                                                                                                                                            0.0s
>  => [base 1/1] FROM docker.io/library/node:20.10.0@sha256:8d0f16fe841577f9317ab49011c6d819e1fa81f8d4af7ece7ae0ac815e07ac84                                                                                                                                                   0.0s
>  => [internal] load build context                                                                                                                                                                                                                                            0.8s
>  => => transferring context: 1.63MB                                                                                                                                                                                                                                          0.8s
>  => CACHED [builder 1/4] WORKDIR /app                                                                                                                                                                                                                                        0.0s
>  => [runner 2/6] RUN addgroup --system --gid 1001 nodejs                                                                                                                                                                                                                     0.8s
>  => [builder 2/4] RUN npm i -g turbo                                                                                                                                                                                                                                         7.4s
>  => [runner 3/6] RUN adduser --system --uid 1001 nextjs                                                                                                                                                                                                                      0.3s
>  => [builder 3/4] COPY . .                                                                                                                                                                                                                                                   4.9s
>  => [builder 4/4] RUN turbo prune clearance_ui --docker                                                                                                                                                                                                                      0.8s
>  => [installer 2/5] COPY --from=builder /app/out/json/ .                                                                                                                                                                                                                     0.0s
>  => [installer 3/5] RUN npm install                                                                                                                                                                                                                                       1845.6s
>  => => # npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead
>  => => # npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
> ```

It did the same for me as well, but ctrl+c and trying again got through that.

**Comment by mmurto** [comment]
[This](https://github.com/doubleopen-project/dos/commit/5b4fb1aa8d9a5f32e3d6209451aba99541199a70) works, but can't exit the image with ctrl+c. :D

**Comment by mmurto** [comment]
Ctrl+c works with this, and everything seems good. Didn't try with env variables. https://github.com/doubleopen-project/dos/compare/main...ui-docker-fixes

**Comment by haikoschol** [comment]
Both UI and API Docker builds still hang for me during `RUN npm ci`. Maybe it works in CI. 🤷‍♂️

**Comment by mmurto** [comment]
Closing in favor of #803 to get ownership of the branch.

**Pull request closed without merging**